### DISCUSSION TITLE
Roughness maps: eval_roughness()

### DISCUSSION THREAD

**Discussion by saeedhd96** [mitsuba-renderer/mitsuba3#454] [Feature requests]
Hi,

I had a request to add a method `eval_roughness(si)` similar to `eval_diffuse_reflectance(si)` to the BSDF and BSDFPtr objects. like this in BRDF. I know it might be a lot of work to add all the methods to every BSDF, but I guess in most cases roughness and albedo are two things that people optimize for and thus need to render maps for. If I am not mistaken, currently we can't access roughness even through the AOV integrators.

**Comment by saeedhd96** [comment]
Would this be possible?

**Reply by wjakob** [reply]
I think we would be willing to entertain such a feature. However, no current projects depend on it, so it's difficult to justify spending time on this. Would you be willing to implement this feature and add a PR including tests for all BRDFs?

**Reply by saeedhd96** [reply]
Sure! I created it!
